specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+  HRESULT Invoke(
+      [in] ICoreWebView2ServiceWorkerRegistration* sender,
+      [in] IUnknown* args);
+}
+
+[uuid(0c0b03bd-ced2-5518-851a-3d3f71fb5c4b), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistration : IUnknown {
+  /// A string representing a URL that defines a service worker's registration scope. It is relative to the base URL of the application.
+  /// By default, the scope value for a service worker registration is set to the directory where the service worker script is located.
+  ///
+  /// The `scope` parameter will be a fully qualified URL that includes both the origin (scheme, host, and optionally port) and the path.
+  /// If the origin and scope includes an internationalized domain name (IDN), it should be represented in punycode format.
+  ///
+  /// For example, if the service worker script is located at https://example.com/sw.js, the default scope is https://example.com/, and the service worker controls all pages under https://example.com/
+  /// If the scope is set to /app/ when registering the service worker, the scope will be https://example.com/app/.
+  ///
+  /// This corresponds to the `scope` property of the `ServiceWorkerRegistration` object in the DOM.
+  /// For more information, see the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/scope).
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Scope([out, retval] LPWSTR* value);
+
+  /// Add an event handler for the `Destroyed` event that is raised when the worker registration is
+  /// unregistered using the JS api `registration.unregister()` method or when the `CoreWebView2ServiceWorkerRegistration`
+  /// object is destroyed. See the
+  /// [Unregister](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/unregister)
+  /// for more information.
+
+  HRESULT add_Destroyed(
+      [in] ICoreWebView2ServiceWorkerRegistrationDestroyedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with `add_Destroyed`.
+  HRESULT remove_Destroyed(
+      [in] EventRegistrationToken token);
+
+
+  /// The active service worker that was created. If there is no active service worker, the completed handler immediately returns `S_OK` and with a null pointer.
+  /// The active service worker is the service worker that controls the pages within the scope of the registration. See the [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker)
+  /// for more information.
+  ///
+  /// This corresponds to the `active` property of the `ServiceWorkerRegistration` object in the DOM.
+  /// For more information, see the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/active).
+  HRESULT GetServiceWorker(
```

Comment:
Should this be a property instead of a method? And should it be named 'ActiveServiceWorker' to sort of match the DOM name?